server/events/project_command_context_builder.go

```diff
@@ -78,6 +84,35 @@ func (cb *CommandScopedStatsProjectCommandContextBuilder) BuildProjectContext(
 	return
 }
 
+// PullReqStatusProjectCommandContextBuilder maps the command context to project command context
+type PullReqStatusProjectCommandContextBuilder struct {
+	ProjectCommandContextBuilder
+}
```

Comment:
Having this is probably a bit overkill, we usually use this pattern primarily to not interfere with upstream code so that merging is easier. In this case, PullRequestStatus is a good optimization that we should upstream without any of the additional SQLocking we've built for it.

Reply:
Hmm I was thinking we didn't want to upstream the sqlocked logic so thought layering it would be a good idea. But it makes sense to upstream the PullRequestStatus without the SQLocking stuff. I've added this one-to-one mapping logic to DefaultProjectCommandContextBuilder and removed this interface entirely.